### DISCUSSION TITLE
Cannot install on ubuntu 22.04

### DISCUSSION THREAD

**Discussion by aleksejs-fomins** [pdfarranger/pdfarranger#977] [Q&A]
I have followed the instructions to install pdfarranger from source on Ubuntu 22.04, Python 3.10.9

When I run

```
pip3 install --user --upgrade https://github.com/pdfarranger/pdfarranger/zipball/main
Collecting https://github.com/pdfarranger/pdfarranger/zipball/main
```

The installation procedure crashes with 
```

  × Running setup.py install for pdfarranger did not run successfully.
  │ exit code: 1
  ╰─> [20 lines of output]
      running install
      /opt/anaconda3/lib/python3.10/site-packages/setuptools/command/install.py:34: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.
        warnings.warn(
      running build
      running build_py
      creating build/lib
      creating build/lib/pdfarranger
      copying pdfarranger/metadata.py -> build/lib/pdfarranger
      copying pdfarranger/iconview.py -> build/lib/pdfarranger
      copying pdfarranger/__init__.py -> build/lib/pdfarranger
      copying pdfarranger/__main__.py -> build/lib/pdfarranger
      copying pdfarranger/exporter.py -> build/lib/pdfarranger
      copying pdfarranger/splitter.py -> build/lib/pdfarranger
      copying pdfarranger/pageutils.py -> build/lib/pdfarranger
      copying pdfarranger/undo.py -> build/lib/pdfarranger
      copying pdfarranger/core.py -> build/lib/pdfarranger
      copying pdfarranger/pdfarranger.py -> build/lib/pdfarranger
      copying pdfarranger/config.py -> build/lib/pdfarranger
      running build_i18n
      error: [Errno 2] No such file or directory: 'msgfmt'
      [end of output]

```

**Comment by aleksejs-fomins** [comment]
Ok, apparently it was sufficient to run

`sudo apt install gettext`

However, that is not self-explanatory from the installation instructions on the main page.

Now, I have another problem. How do I actually run the code??

I have attempted to run it using `python3 -m pdfarranger`, that crashed with 

`ModuleNotFoundError: No module named 'gi'`

I proceeded to do the following 

```
sudo apt install libcairo2-dev libxt-dev libgirepository1.0-dev
pip install pycairo PyGObject
```

Now when I run `python3 -m pdfarranger` I get

'xkbcommon: ERROR: failed to add default include path /opt/anaconda3/share/X11/xkb'

And the program crashes with a segmentation fault. What am I doing wrong?

**Comment by jeromerobert** [comment]
> `ModuleNotFoundError: No module named 'gi'`

It's provided by `python3-gi` that you should have installed if you followed the `README`.

* See https://github.com/pdfarranger/pdfarranger#install-from-source (Debian based distribution)
* Yes `gettext` is missing. Thanks I'll add it.
* Mixing `pip`, `conda` and `apt` will give you indigestion. Uninstall `pycairo`,  `PyGObject`, ensure you have no conda stuff in your environment and follow the `README`.
* If you use virtualenv or kind of do not forget `--system-site-packages`
* If you want to install with conda, you'll be on your own, because to my knowledge this has never been done (but if you manage to do it, share it!).

**Reply by aleksejs-fomins** [reply]
I see, indeed the problem was likely caused by using anaconda python by default. I'm sorry, it was not obvious to me that it was not ok to use anaconda-based python. I did indeed follow the readme before and installed `python3-gi`, but it was not visible to the anaconda python for whatever reason.
